*** OldStyleMemberships.py.orig	2008-01-15 01:57:37.000000000 -0600
--- OldStyleMemberships.py	2008-11-20 15:31:31.000000000 -0600
***************
*** 30,35 ****
--- 30,36 ----
  from Mailman import Utils
  from Mailman import Errors
  from Mailman import MemberAdaptor
+ from SELSdeleteuser import *
  
  ISREGULAR = 1
  ISDIGEST = 2
***************
*** 229,234 ****
--- 230,237 ----
              dict = getattr(self.__mlist, attr)
              if dict.has_key(memberkey):
                  del dict[memberkey]
+ 	# Remove the SELS keys for the member. Removekeys method is defined in SELSdeleteuser.py
+ 	Removekeys(self.__mlist.internal_name(), member)
  
      def changeMemberAddress(self, member, newaddress, nodelete=0):
          assert self.__mlist.Locked()
***************
*** 252,257 ****
--- 255,262 ----
          # Set the entire options bitfield
          if flags:
              self.__mlist.user_options[newaddress.lower()] = flags
+ 	# Remove the SELS keys for the member. Removekeys method is defined in SELSdeleteuser.py
+         Removekeys(self.__mlist.internal_name(), member)
  
      def setMemberPassword(self, memberkey, password):
          assert self.__mlist.Locked()
